Yes because I just tested and it fails:
```python
    f = oms.String("foo")
    repr(f)
```

```
Traceback (most recent call last):
  File "/Users/julianuspfeuffer/git/OpenMS/cmake-build-Release-Clang_15.0.0_arm64-apple-darwin23.0.0/pyOpenMS/test.py", line 8, in <module>
    repr(f)
TypeError: __repr__ returned non-string (type bytes)
```

--- a/src/pyOpenMS/addons/String.pyx
+++ b/src/pyOpenMS/addons/String.pyx
@@ -39,7 +39,7 @@
         return self.toString()
 
     def __repr__(self):
-        return self.c_str()
+        return f"String('{self.toString()}')"
 
     # TODO does this really need to be callable from python? How about cdef?
     def c_str(self):

